llvm/include/llvm/IR/Type.h

```diff
@@ -184,7 +204,7 @@ class Type {
   /// Return true if this is one of the floating-point types
   bool isFloatingPointTy() const {
     return isIEEELikeFPTy() || getTypeID() == X86_FP80TyID ||
-           getTypeID() == PPC_FP128TyID;
+           getTypeID() == PPC_FP128TyID || isDecimalFloatingPointTy();
   }
```

Comment:
We'll need to audit (separately from this review) the existing uses of `isFloatingPointTy()` to determine if, for each case, whether the code will need changes to handle both binary and decimal FP types. I filed issue #34 to follow up on this.

Reply:
Agree. I can work on that eventually.